### PR TITLE
Refactor to allow Murfey servers to operate for any instrument in configuration

### PR DESCRIPTION
**Pull request by d-j-hatton**
Also allows one server to use another for auth if configured. This will allow the use of only a single central server for auth

### CODE REVIEW

**Review by stephen-riggs — APPROVED** [review]

File: src/murfey/instrument_server/api.py [hunk 1]
```diff
@@ -199,22 +216,30 @@ class ProcessingParameterBlock(BaseModel):
     params: ProcessingParameters
 
 
-@router.post("/processing_parameters")
-def register_processing_parameters(proc_param_block: ProcessingParameterBlock):
+@router.post("/sessions/{session_id}/processing_parameters")
+def register_processing_parameters(
+    session_id: MurfeySessionID, proc_param_block: ProcessingParameterBlock
```
[the remark this answers is not in the document]
Comment (d-j-hatton) [reply src/murfey/instrument_server/api.py:221]: It just uses it for auth. The `MurfeySessionID` contains a `fastapi` `Dependency` which validates a token against the Murfey session

File: src/murfey/server/api/__init__.py [hunk 2]
```diff
@@ -1596,20 +1769,28 @@ def change_monitoring_status(visit_name: str, on: int):
...
         "json": post_info.data,
     }
     if _transport_object:
+        machine_config = get_machine_config(instrument_name=instrument_name)[
+            instrument_name
+        ]
         _transport_object.send(machine_config.feedback_queue, zocalo_message)
 
 
 @router.get("/visits/{session_id}/upstream_visits")
```
[the remark this answers is not in the document]
Comment (d-j-hatton) [reply src/murfey/server/api/__init__.py:1786]: Agreed. Will see how simple it is to change

File: src/murfey/client/__init__.py [hunk 3]
```diff
@@ -100,6 +100,7 @@ def _check_for_updates(
 
 def run():
     config = read_config()
+    instrument_name = config["Murfey"]["instrument_name"]
```
[the remark this answers is not in the document]
Comment (d-j-hatton) [reply src/murfey/client/__init__.py:103]: Don't really know, the config key is about as old as the codebase

File: src/murfey/server/api/__init__.py [hunk 4]
```diff
@@ -1162,15 +1280,23 @@ async def request_tomography_preprocessing(
...
+        db.exec(select(Session).where(Session.id == session_id)).one().instrument_name
+    )
+    machine_config = get_machine_config(instrument_name=instrument_name)[
+        instrument_name
+    ]
     check_path = (
         machine_config.rsync_basepath / base_path
         if machine_config
-        else Path(f"/dls/{get_microscope(machine_config=machine_config)}") / base_path
+        else Path(f"/dls/{instrument_name}") / base_path
```
[the remark this answers is not in the document]
Comment (d-j-hatton) [reply src/murfey/server/api/__init__.py:1299]: Yes. It's only used if there is no machine configuration but should probably just throw an exception in that case

File: src/murfey/server/api/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (d-j-hatton) [reply src/murfey/server/api/__init__.py:1786]: This one was actually just an error

**Pull request merged**